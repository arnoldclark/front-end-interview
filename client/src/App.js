import React from "react";
import { Navbar } from "./components/Navbar";

const App = () => {
  return (
    <>
      <Navbar />

      <div className="ch-container ch-mt--4">
        <p>Let&apos;s get started!</p>
      </div>
    </>
  );
};

export default App;
