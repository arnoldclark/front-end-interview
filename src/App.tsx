import React from "react";
import { Navbar } from "./components/Navbar";
import CarCard from "./components/CarCard";
import { CarForm } from "./components/CarForm";
import useCars from "./Hooks/useCars";

const App = () => {
  const carsQuery = useCars();

  return (
    <>
      <Navbar />

      <div className="ch-container ch-mt--4">
        <CarForm />

        {carsQuery.isLoading && <p>Loading</p>}

        {carsQuery.isError && <p>{carsQuery.error.message}</p>}

        {
          /* Loop over the cars[] array and output some cards */
          carsQuery.data &&
            carsQuery.data.map((item) => {
              return (
                <CarCard
                  key={item.reg}
                  make={item.make}
                  model={item.model}
                ></CarCard>
              );
            })
        }
      </div>
    </>
  );
};

export default App;
