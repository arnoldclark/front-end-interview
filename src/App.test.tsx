import { render } from "@testing-library/react";
import App from "./App";

it("should match the snapshot", () => {
  const result = render(<App />);
  expect(result.container).toMatchSnapshot();
});
