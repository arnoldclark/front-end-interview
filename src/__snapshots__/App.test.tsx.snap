// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<div>
  <header
    class="ch-navbar"
  >
    <div
      class="ch-container"
    >
      <div
        class="ch-row"
      >
        <div
          class="xs:ch-col--12 sm:ch-col--3"
        >
          <a
            class="ch-navbar__title"
            href="/"
          >
            Arnold Clark Garage
          </a>
        </div>
      </div>
    </div>
  </header>
  <div
    class="ch-container ch-mt--4"
  >
    <p>
      Let's get started!
    </p>
  </div>
</div>
`;
